src/table/hooks/use-column-widths.ts

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import { Column } from '../../types';
+import { MIN_COLUMN_WIDTH, ColumnWidthTypes } from '../constants';
+import useDidUpdateEffect from './use-did-update-effect';
+import useMeasureText, { MeasureTextHook } from '../virtualized-table/hooks/use-measure-text';
+
+/**
+ * Calculates column widths in pixels, based on column settings and the table width.
+ * First, pixel values for the three independent types ('pixels', 'percentage', 'hug') are set.
+ * Then the remaining width is divided equally between the 'fill' columns, if there is any
+ * The widths are sorted in the order they will be displayed
+ */
+export const getColumnWidths = (
+  columns: Column[],
+  tableWidth: number,
+  { measureText, estimateWidth }: MeasureTextHook
+) => {
+  if (!columns?.length || tableWidth === 0) return [];
+
+  const columnWidths: number[] = [];
+  const fillColumnIndexes: number[] = [];
+  let sumFillWidths = tableWidth;
+
+  columns.forEach((col, idx) => {
+    if (col.columnWidth) {
+      const {
+        columnWidth: { type, pixels, percentage },
+        label,
+        qApprMaxGlyphCount,
+      } = col;
+      let newWidth = 0;
+
+      const addKnownWidth = () => {
+        columnWidths[idx] = Math.max(MIN_COLUMN_WIDTH, newWidth);
+        sumFillWidths -= columnWidths[idx];
+      };
+
+      switch (type) {
+        case ColumnWidthTypes.PIXELS:
+          if (!pixels) throw new Error(`pixel value is not defined`);
```

Comment:
Why are you throwing an error here? This seems like it would case issue for any existing sn-table objects (if there are any). If an existing table switches to pixel value via PP this would throw and the chart fails to render.

Reply:
it was, and I fixed it on the branch where we tested yesterday. I have a commit incomming with it